### PR TITLE
Remove actions

### PR DESCRIPTION
**Pull request by kosteman**

### CODE REVIEW

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java [hunk 1]
```diff
@@ -85,6 +88,55 @@ public void reInitPage() {
...
+    @And("^пользователь проверяет что текст \"(.*?)\" отсутствует на странице$")
+    public void checkTextIsNotPresent(String text) {
+        super.checkTextIsNotPresent(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^пользователь проверяет что модальное окно с текстом \"(.*?)\" появляется$")
+    public void checkModalWindowAppears(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47EwrFHbA8-Egpm_&open=AWZh47EwrFHbA8-Egpm_)

[ISSUE_KEY]: AWZh47EwrFHbA8-Egpm_

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java [hunk 2]
```diff
@@ -84,4 +86,53 @@ public void goToPageByUrl(String url) throws PageInitializationException {
     public void reInitPage() {
         super.reInitPage();
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^user accepts alert with text \"(.*?)\"$")
+    public void acceptAlert(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47DWrFHbA8-Egpm1&open=AWZh47DWrFHbA8-Egpm1)

[ISSUE_KEY]: AWZh47DWrFHbA8-Egpm1

File: plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java [hunk 3]
```diff
@@ -3,10 +3,9 @@
 import ru.sbtqa.tag.pagefactory.jdi.utils.JDIUtils;
 
 /**
- * Contains basic ru.sbtqa.tag.pagefactory.mobile.actions in particular with jdi elements
- * If we want to extend this functional - inherit from this class
+ * Inherit your jdi page objects from this class
  */
-public abstract class JDIPage extends DefaultPage {
+public abstract class JDIPage implements Page {
```
Comment (sbtqasonar) [inline plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java]: Code Smell Code Smell: Convert the abstract class "JDIPage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47E7rFHbA8-EgpnB&open=AWZh47E7rFHbA8-EgpnB)

[ISSUE_KEY]: AWZh47E7rFHbA8-EgpnB

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java [hunk 4]
```diff
@@ -1,39 +1,8 @@
...
-     * @param text text on page to swipe to
-     * @throws SwipeException if the text is not found
-     */
-    @ActionTitle("ru.sbtqa.tag.pagefactory.swipe.match")
-    public void swipeToTextByMatch(String strategy, String text) throws SwipeException {
-        SwipeUtils.swipeToText(MatchStrategy.valueOf(strategy), text);
-    }
-}
\ No newline at end of file
+public abstract class MobilePage implements Page {}
```
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java]: Code Smell Code Smell: Convert the abstract class "MobilePage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46sarFHbA8-Egpmw&open=AWZh46sarFHbA8-Egpmw)

[ISSUE_KEY]: AWZh46sarFHbA8-Egpmw

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java [hunk 5]
```diff
@@ -84,4 +86,53 @@ public void goToPageByUrl(String url) throws PageInitializationException {
...
+    @And("^user checks that text \"(.*?)\" appears on the page$")
+    public void checkTextAppears(String text) throws WaitException {
+        super.checkTextAppears(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^user checks that text \"(.*?)\" is absent on the page$")
+    public void checkTextIsNotPresent(String text) {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47DWrFHbA8-Egpm4&open=AWZh47DWrFHbA8-Egpm4)

[ISSUE_KEY]: AWZh47DWrFHbA8-Egpm4

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java [hunk 6]
```diff
@@ -84,4 +86,53 @@ public void goToPageByUrl(String url) throws PageInitializationException {
...
+        super.checkTextIsNotPresent(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^user checks that modal window with text \"(.*?)\" is appears$")
+    public void checkModalWindowAppears(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47DWrFHbA8-Egpm5&open=AWZh47DWrFHbA8-Egpm5)

[ISSUE_KEY]: AWZh47DWrFHbA8-Egpm5

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java [hunk 7]
```diff
@@ -85,6 +88,55 @@ public void reInitPage() {
...
+    @And("^пользователь принимает уведомление с текстом \"(.*?)\"$")
+    public void acceptAlert(String text) throws WaitException {
+        super.acceptAlert(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^пользователь отклоняет уведомление с текстом \"(.*?)\"$")
+    public void dismissAlert(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47EwrFHbA8-Egpm8&open=AWZh47EwrFHbA8-Egpm8)

[ISSUE_KEY]: AWZh47EwrFHbA8-Egpm8

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java [hunk 8]
```diff
@@ -84,4 +86,53 @@ public void goToPageByUrl(String url) throws PageInitializationException {
...
+    @And("^user dismisses alert with text \"(.*?)\"$")
+    public void dismissAlert(String text) throws WaitException {
+        super.dismissAlert(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^user checks that text \"(.*?)\" appears on the page$")
+    public void checkTextAppears(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47DWrFHbA8-Egpm3&open=AWZh47DWrFHbA8-Egpm3)

[ISSUE_KEY]: AWZh47DWrFHbA8-Egpm3

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/MobileSteps.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import cucumber.api.java.Before;
+import cucumber.api.java.en.And;
+import ru.sbtqa.tag.pagefactory.exceptions.SwipeException;
+import ru.sbtqa.tag.stepdefs.MobileGenericSteps;
+
+public class MobileSteps extends MobileGenericSteps {
+
+    @Before(order = 1)
+    public void initMobile() {
```
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/MobileSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46xKrFHbA8-Egpmx&open=AWZh46xKrFHbA8-Egpmx)

[ISSUE_KEY]: AWZh46xKrFHbA8-Egpmx

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java [hunk 10]
```diff
@@ -85,6 +88,55 @@ public void reInitPage() {
...
+    public void checkModalWindowAppears(String text) throws WaitException {
+        super.checkModalWindowAppears(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^пользователь проверяет что элемент с текстом \"(.*?)\" существует$")
+    @When("^пользователь проверяет что текст \"(.*?)\" отоброжается$")
+    public void checkElementWithTextIsPresent(String text) {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47EwrFHbA8-EgpnA&open=AWZh47EwrFHbA8-EgpnA)

[ISSUE_KEY]: AWZh47EwrFHbA8-EgpnA

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java [hunk 11]
```diff
@@ -85,6 +88,55 @@ public void reInitPage() {
...
+    @And("^пользователь отклоняет уведомление с текстом \"(.*?)\"$")
+    public void dismissAlert(String text) throws WaitException {
+        super.dismissAlert(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^пользователь проверяет что текст \"(.*?)\" появляется на странице$")
+    public void checkTextAppears(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47EwrFHbA8-Egpm9&open=AWZh47EwrFHbA8-Egpm9)

[ISSUE_KEY]: AWZh47EwrFHbA8-Egpm9

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java [hunk 12]
```diff
@@ -84,4 +86,53 @@ public void goToPageByUrl(String url) throws PageInitializationException {
...
+    @And("^user accepts alert with text \"(.*?)\"$")
+    public void acceptAlert(String text) throws WaitException {
+        super.acceptAlert(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^user dismisses alert with text \"(.*?)\"$")
+    public void dismissAlert(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47DWrFHbA8-Egpm2&open=AWZh47DWrFHbA8-Egpm2)

[ISSUE_KEY]: AWZh47DWrFHbA8-Egpm2

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/WebGenericSteps.java [hunk 13]
```diff
@@ -8,8 +8,13 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import ru.sbtqa.tag.pagefactory.PageManager;
+import ru.sbtqa.tag.pagefactory.annotations.ActionTitle;
+import ru.sbtqa.tag.pagefactory.annotations.ActionTitles;
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/WebGenericSteps.java]: Code Smell Code Smell: Remove this unused import 'ru.sbtqa.tag.pagefactory.annotations.ActionTitles'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47BsrFHbA8-Egpm0&open=AWZh47BsrFHbA8-Egpm0)

[ISSUE_KEY]: AWZh47BsrFHbA8-Egpm0

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/MobileSteps.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+import cucumber.api.java.Before;
+import cucumber.api.java.en.And;
+import ru.sbtqa.tag.pagefactory.exceptions.SwipeException;
+import ru.sbtqa.tag.stepdefs.MobileGenericSteps;
+
+public class MobileSteps extends MobileGenericSteps {
+
+    @Before(order = 1)
+    public void initMobile() {
```
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/MobileSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46zKrFHbA8-Egpmy&open=AWZh46zKrFHbA8-Egpmy)

[ISSUE_KEY]: AWZh46zKrFHbA8-Egpmy

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java [hunk 15]
```diff
@@ -84,4 +86,53 @@ public void goToPageByUrl(String url) throws PageInitializationException {
...
+    public void checkModalWindowAppears(String text) throws WaitException {
+        super.checkModalWindowAppears(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^user checks that element with text \"(.*?)\" is present$")
+    @When("^user checks that the text \"(.*?)\" is visible$")
+    public void checkElementWithTextIsPresent(String text) {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47DWrFHbA8-Egpm6&open=AWZh47DWrFHbA8-Egpm6)

[ISSUE_KEY]: AWZh47DWrFHbA8-Egpm6

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java [hunk 16]
```diff
@@ -85,6 +88,55 @@ public void reInitPage() {
...
+        super.checkTextAppears(text);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @And("^пользователь проверяет что текст \"(.*?)\" отсутствует на странице$")
+    public void checkTextIsNotPresent(String text) {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47EwrFHbA8-Egpm-&open=AWZh47EwrFHbA8-Egpm-)

[ISSUE_KEY]: AWZh47EwrFHbA8-Egpm-

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java [hunk 17]
```diff
@@ -85,6 +88,55 @@ public void reInitPage() {
         super.reInitPage();
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @And("^пользователь принимает уведомление с текстом \"(.*?)\"$")
+    public void acceptAlert(String text) throws WaitException {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/WebSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47EwrFHbA8-Egpm7&open=AWZh47EwrFHbA8-Egpm7)

[ISSUE_KEY]: AWZh47EwrFHbA8-Egpm7

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java [hunk 18]
```diff
@@ -3,19 +3,12 @@
...
-import ru.sbtqa.tag.pagefactory.web.utils.WebExpectedConditionsUtils;
-import ru.sbtqa.tag.qautils.errors.AutotestError;
 
 /**
- * Contains basic ru.sbtqa.tag.pagefactory.mobile.actions in particular with web elements
- * If we want to extend this functional - inherit from this class
+ * Inherit your web page objects from this class
  */
-public abstract class WebPage extends DefaultPage {
+public abstract class WebPage implements Page {
```
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java]: Code Smell Code Smell: Convert the abstract class "WebPage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46-jrFHbA8-Egpmz&open=AWZh46-jrFHbA8-Egpmz)

[ISSUE_KEY]: AWZh46-jrFHbA8-Egpmz

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/WebGenericSteps.java (hunk 13 above)
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/stepdefs/WebGenericSteps.java]: Code Smell Code Smell: Remove this unused import 'ru.sbtqa.tag.pagefactory.annotations.ActionTitles'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47BsrFHbA8-Egpm0&open=AWZh47BsrFHbA8-Egpm0)

[ISSUE_KEY]: AWZh47BsrFHbA8-Egpm0

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/MobileSteps.java (hunk 9 above)
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/en/MobileSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46xKrFHbA8-Egpmx&open=AWZh46xKrFHbA8-Egpmx)

[ISSUE_KEY]: AWZh46xKrFHbA8-Egpmx

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java (hunk 4 above)
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java]: Code Smell Code Smell: Convert the abstract class "MobilePage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46sarFHbA8-Egpmw&open=AWZh46sarFHbA8-Egpmw)

[ISSUE_KEY]: AWZh46sarFHbA8-Egpmw

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java (hunk 18 above)
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java]: Code Smell Code Smell: Convert the abstract class "WebPage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46-jrFHbA8-Egpmz&open=AWZh46-jrFHbA8-Egpmz)

[ISSUE_KEY]: AWZh46-jrFHbA8-Egpmz

File: plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java (hunk 3 above)
Comment (sbtqasonar) [inline plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java]: Code Smell Code Smell: Convert the abstract class "JDIPage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47E7rFHbA8-EgpnB&open=AWZh47E7rFHbA8-EgpnB)

[ISSUE_KEY]: AWZh47E7rFHbA8-EgpnB

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/MobileSteps.java (hunk 14 above)
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/stepdefs/ru/MobileSteps.java]: Code Smell Code Smell: Add the "@Override" annotation above this method signature ([squid:S1161](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1161))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46zKrFHbA8-Egpmy&open=AWZh46zKrFHbA8-Egpmy)

[ISSUE_KEY]: AWZh46zKrFHbA8-Egpmy

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java (hunk 4 above)
Comment (sbtqasonar) [inline plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java]: Code Smell Code Smell: Convert the abstract class "MobilePage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46sarFHbA8-Egpmw&open=AWZh46sarFHbA8-Egpmw)

[ISSUE_KEY]: AWZh46sarFHbA8-Egpmw

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java (hunk 18 above)
Comment (sbtqasonar) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java]: Code Smell Code Smell: Convert the abstract class "WebPage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh46-jrFHbA8-Egpmz&open=AWZh46-jrFHbA8-Egpmz)

[ISSUE_KEY]: AWZh46-jrFHbA8-Egpmz

File: plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java (hunk 3 above)
Comment (sbtqasonar) [inline plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java]: Code Smell Code Smell: Convert the abstract class "JDIPage" into an interface. ([squid:S1610](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1610))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=96&issues=AWZh47E7rFHbA8-EgpnB&open=AWZh47E7rFHbA8-EgpnB)

[ISSUE_KEY]: AWZh47E7rFHbA8-EgpnB

File: plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java (hunk 4 above)
Comment (kosteman) [reply plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/MobilePage.java]: Нет, нам нужны конструкторы

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java (hunk 18 above)
Comment (kosteman) [reply plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebPage.java]: Нет, нам нужны конструкторы

File: plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java (hunk 3 above)
Comment (kosteman) [reply plugins/jdi-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/JDIPage.java]: Нет, нам нужны конструкторы

File: page-factory-api/src/main/java/ru/sbtqa/tag/pagefactory/Page.java [hunk 19]
```diff
@@ -1,10 +1,15 @@
 package ru.sbtqa.tag.pagefactory;
 
+import ru.sbtqa.tag.pagefactory.annotations.PageEntry;
 
-/**
- * Entry point to the page-factory
- */
```
Comment (clicman) [inline page-factory-api/src/main/java/ru/sbtqa/tag/pagefactory/Page.java]: Why you've removed the description?

**Pull request merged**